core/typedefs.h

```diff
@@ -154,7 +154,23 @@ inline bool is_power_of_2(const T x) {
 }
 
 // Function to find the next power of 2 to an integer.
-static _FORCE_INLINE_ unsigned int next_power_of_2(unsigned int x) {
+constexpr uint64_t next_power_of_2(uint64_t x) {
+	if (x == 0) {
+		return 0;
+	}
+
+	--x;
+	x |= x >> 1;
+	x |= x >> 2;
+	x |= x >> 4;
+	x |= x >> 8;
+	x |= x >> 16;
+	x |= x >> 32;
+
+	return ++x;
+}
```

Comment:
As long as we're here, might as well make sure we're using the proper argument syntax. Applies to all functions touched in this script.
```suggestion
constexpr uint64_t next_power_of_2(uint64_t p_x) {
	if (p_x == 0) {
		return 0;
	}

	--p_x;
	p_x |= p_x >> 1;
	p_x |= p_x >> 2;
	p_x |= p_x >> 4;
	p_x |= p_x >> 8;
	p_x |= p_x >> 16;
	p_x |= p_x >> 32;

	return ++p_x;
}
```

Reply:
Good point, but I went with `p_number`, since `x` can imply a horizontal coordinate. I was going to use `p_num` at first but `nearest_shift` uses `p_number` already so I suppose it's less friction to use `p_number`, but if `p_num` is desired I could change it to that.